src/platform/Tizen/DnssdImpl.cpp

```diff
@@ -356,17 +342,13 @@ void OnResolve(dnssd_error_e result, dnssd_service_h service, void * userData)
 
     rCtx->mResult.mAddress.emplace(ipAddr);
 
-    {
-        // Before calling the Resolve() callback, we need to lock stack mutex.
-        // However, we cannot lock the stack mutex from here, because we might
-        // face lock inversion problem. This callback (OnResolve()) is called
-        // with the NSD internal mutex locked, which is also locked by the
-        // dnssd_create_remote_service() function called in the Resolve(), and
-        // the Resolve() itself is called with the stack mutex locked.
-        chip::GAutoPtr<GSource> sourceIdle(g_idle_source_new());
-        g_source_set_callback(sourceIdle.get(), OnResolveFinalize, rCtx, NULL);
-        g_source_attach(sourceIdle.get(), g_main_context_get_thread_default());
-    }
+    err = chip::DeviceLayer::SystemLayer().ScheduleLambda([rCtx] {
+        ChipLogProgress(DeviceLayer, "DNSsd Handle resolve task on schedule lambda");
+
+        rCtx->Finalize(CHIP_NO_ERROR);
+        rCtx->mInstance->RemoveContext(rCtx);
+    });
+    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(DeviceLayer, "Failed to schedule resolve task: %s", err.AsString()));
```

Comment:
Use ` %" CHIP_ERROR_FORMAT, err.Format()));`

Reply:
Updated it.